Sources/StreamVideo/Utils/Sorting.swift

```diff
@@ -4,110 +4,230 @@
+/// it moves to the next comparator, and so on, until a decision can be made or all comparators have been exhausted.
+public func combineComparators<T>(_ comparators: [StreamSortComparator<T>]) -> StreamSortComparator<T> {
+    return { a, b in
+        for comparator in comparators {
+            let result = comparator(a, b)
+            if result != .orderedSame {
+                return result
+            }
+        }
+        return .orderedSame
+    }
 }
 
-public var roles: StreamSortComparator<CallParticipant> = { (p1, p2) in
-    if p1.roles == p2.roles { return .orderedSame }
-    let prioRoles = ["admin", "host", "speaker"]
-    for role in prioRoles {
-        if p1.roles.contains(role) && !p2.roles.contains(role) {
-            return .orderedDescending
-        }
-        if p2.roles.contains(role) && !p1.roles.contains(role) {
-            return .orderedAscending
+/// Returns a new comparator that only applies the given comparator if the predicate returns true.
+/// If the predicate returns false, it deems the two elements "equal" (i.e., orderedSame).
+public func conditional<T>(_ predicate: @escaping (T, T) -> Bool) -> (@escaping StreamSortComparator<T>) -> StreamSortComparator<T> {
+    return { comparator in
+        return { a, b in
+            if !predicate(a, b) {
+                return .orderedSame
+            }
+            return comparator(a, b)
         }
     }
-    return .orderedSame
 }
 
-public var userId: StreamSortComparator<CallParticipant> = { (p1, p2) in
-    p1.id >= p2.id ? .orderedDescending : .orderedAscending
+/// Returns a comparator that always deems two elements "equal" regardless of their actual values.
+public func noop<T>() -> StreamSortComparator<T> { { _, _ in .orderedSame } }
+
+/// A specific conditional comparator for CallParticipant that checks if either participant's track is not visible.
+public let ifInvisible = conditional { (rhs: CallParticipant, lhs: CallParticipant) -> Bool in
+    !rhs.showTrack || !lhs.showTrack
 }
 
-public var joinedAt: StreamSortComparator<CallParticipant> = { (p1, p2) in
-    p1.joinedAt <= p2.joinedAt ? .orderedDescending : .orderedAscending
+// MARK: Instance
+
+/// Comparator which sorts participants by the fact that they are the dominant speaker or not.
+public var dominantSpeaker: StreamSortComparator<CallParticipant> = { comparison($0, $1, keyPath: \.isDominantSpeaker) }
+
+/// Comparator which sorts participants by the fact that they are speaking or not.
+public var isSpeaking: StreamSortComparator<CallParticipant> = { comparison($0, $1, keyPath: \.isSpeaking) }
+
+/// Comparator which prioritizes participants who are pinned.
+public var pinned: StreamSortComparator<CallParticipant> = { a, b in
+    switch (a.pin, b.pin) {
+    case (nil, _?): return .orderedDescending
+    case (_?, nil): return .orderedAscending
+    case (let aPin?, let bPin?) where !aPin.isLocal && bPin.isLocal: return .orderedAscending
```

Comment:
shouldn't this be the opposite? Remote pins have priority over local ones.

Reply:
You are right! I updated it and added the missing test case to validate it.